### PR TITLE
scripts/unpack: save bandwidth/time using git pull [WIP]

### PR DESCRIPTION
**Pull request by ToKe79**
For packages where we use a git repository instead of sources archive, we do not need to clone the whole repository every time.

At the beginning we must source the package.mk to have a valid PKG_GIT_URL to skip the cleaning of the old version. Then, instead of doing a `git clone` we:
- find the previous package build folder in BUILD (by comparing PKG_NAME with STAMP_PKG_NAME in .libreelec-unpack and then checking if PKG_GIT_URL is in .git/config) -- this confirms that we are in the correct folder AND the folder is a cloned repository
- remove all changes in the folder - `git clean -fdx` cleans all untracked files and folders and `git checkout -- .` removes all changes to original files (e.g. when patching some files)
- then we must switch to an existing branch - `git checkout master` so we can
- pull the changes from the GitHub repository - `git pull`

If such folder is not found, we use the clean script to clean any old build folders of that package and clone the repository.

We checkout the commit which we want to compile using `git checkout $PKG_VERSION` as previously (this is not changed). The GIT_VERSION for each individual libretro core is parsed in one of the Makefiles of each core. This is also not changed, so cores will build with the GIT_VERSION information.

This is first concept with some debug messages and no indentation changes for easier tracking of the changes.

### CODE REVIEW

**Comment by kivutar** [comment]
Sorry for the hard work you put on this @ToKe79 but I prefer to avoid huge changes to `scripts/` as it makes merging upstream hazardous.

Also, a more complex implementation like this is also more prone to errors. For example, sourcing package.mk can cause variable name collisions during an upstream merge.

I'm going to wait for the reviews of other contributors.

File: scripts/unpack [hunk 1]
```diff
@@ -103,9 +108,55 @@ if [ -d "$SOURCES/$1" -o -d "$PKG_DIR/sources" -o -n "$PKG_GIT_URL" ]; then
...
+        if [ -f $f/.libreelec-unpack ]; then
+          echo "**** debug: '.libreelec-unpack' found"
+          source $f/.libreelec-unpack
+          echo "**** debug: STAMP_PKG_NAME=$STAMP_PKG_NAME"
+          if [ "$PKG_NAME" = "$STAMP_PKG_NAME" ]; then
+            echo "**** debug: NAME matched"
+            if [ -f $f/.git/config ]; then
+              echo "**** debug: .git/config found:"
+              cat $f/.git/config
+              GIT_MATCH=$(grep "$PKG_GIT_URL" $f/.git/config | wc -l)
```
Comment (RobLoach) [inline scripts/unpack]: Rather than parsing .git/config, you could likely use `git remote get-url origin`

File: scripts/unpack [hunk 2]
```diff
@@ -103,9 +108,55 @@ if [ -d "$SOURCES/$1" -o -d "$PKG_DIR/sources" -o -n "$PKG_GIT_URL" ]; then
...
+      git checkout master
+      # pull changes
+      echo "**** debug: git pull"
+      git pull
+    else
+      # clean
+      $SCRIPTS/clean $1
+      # clone
     git clone $PKG_GIT_URL $BUILD/$PKG_NAME-$PKG_VERSION/
     cd $BUILD/$PKG_NAME-$PKG_VERSION/
```
Comment (RobLoach) [inline scripts/unpack]: Whitespace fix.

**Comment by RobLoach** [comment]
While I like the general idea,  I'm with Kivutar. This seems like something you could push forwards to [LibreELEC](https://github.com/LibreELEC/LibreELEC.tv). I'm sure they'd love to see a sped up git clone process.

**Comment by ToKe79** [comment]
Thank you for you input.

I parse `.git/config` due to the fact, that:
- I make sure that the current folder is a git repository
- and that the url is the same as in the package.mk
In case I would not check for .git/config and just do `git remote get-url origin` and the current folder would not be in a git repository, I could get the url of the Lakka git. But the result would be same - it would not match and I would clean :-)

I also understand that merging this to LE might be complicated (LE to my knowledge does not use `git clone` at all) and maintaining this change locally when merging from LE could and probably will introduce some overhead. But to my defense :-) the original code for `git clone` based on PKG_SECTION is also not merged upstream.

I have checked if sourcing package.mk is really necessary, but it is not, as is already done with this code:
https://github.com/libretro/Lakka-LibreELEC/blob/07b402ca29ce2c32d5df2ef20803f3aacf595a49/scripts/unpack#L21

I can change also the checking of the git url as @RobLoach pointed out - seems more cleaner way . And no hard feelings if this is not going to be merged.

File: scripts/unpack (hunk 2 above)
Comment (ToKe79) [reply scripts/unpack]: On which line was the incorrect whitespace? I cannot see the change.

Comment (RobLoach) [reply scripts/unpack]: Ah, the `git clone` and `cd ...` should be two spaces indented to make sure the whitespace is correct.

Line 157 and 158

**Comment by ToKe79** [comment]
There is git handler in LE9 which solves this.

**Pull request closed without merging**